### PR TITLE
Remove pallet society

### PR DESCRIPTION
**Pull request by rakanalh**
### Description

Seems like a useless pallet that nobody uses. Also requires migrations in v1.0.0 which i would rather remove the need for rather than keeping the pallet.

### Types of Changes

- [x] Tech Debt (Code improvements)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Dependency upgrade (A change in substrate or any 3rd party crate version)

### Migrations and Hooks

- [ ] This change requires a runtime migration.
- [ ] Modifies `on_initialize`
- [ ] Modifies `on_finalize`

### Checklist

- [x] Change has been tested locally.
- [ ] Change adds / updates tests.
- [x] Changelog doc updated.

### CODE REVIEW

File: scripts/pre-commit.sh [hunk 1]
```diff
@@ -8,8 +8,3 @@ if [ $? -ne 0 ]; then
 	exit 1
 fi
 
-dprint check
```
Comment (rakanalh) [inline scripts/pre-commit.sh:11]: UNDO this change.

**Comment by rakanalh** [comment]
I agree, will implement a migration for this.

**Review by aie0 — APPROVED** [review]
